### DISCUSSION TITLE
You can ask any question here.

### DISCUSSION THREAD

**Discussion by pykt-team** [pykt-team/pykt-toolkit#30] [Q&A] [answered]

**Comment by 123kristin** [comment]
Thanks to the pykt library for your great contribution to the knowledge tracing field!. What is the parameter ‘emb_type’ in the At-dkt code? It's not the default value 'qid', is it?

**Reply by sonyawong** [reply, marked as answer]
hi~exactly, not all the values of emb_type are set to "qid" in our repo. For more detail about the parameters setting of our model, you can visit https://drive.google.com/drive/folders/1MWYXj73Ke3zC6bm3enu1gxQQKAHb37hz?usp=drive_link.

**Reply by 123kristin** [reply]
You've been a great help

**Reply by 123kristin** [reply]
About rkt model,   How to get phi_array_1_2_3_4.pkl file?

**Comment by linchuanghong** [comment]
What is the meaning of "is_repeat" data in the data set generated after the original data is preprocessed?

**Reply by sonyawong** [reply]
It is a flag to denote whether the interaction is expanded. Since there is a question contains multiple knowledge components (KCs), for those questions, the interaction will be expand based on KC level one by one.

**Comment by cwsy92** [comment]
![Screenshot from 2023-10-04 10-59-38](https://github.com/pykt-team/pykt-toolkit/assets/146905955/d31649af-3bb2-4274-ac6c-259cde6ebc65)
I understand that there are various scenarios e.g. Questions Level (All-in-One), KC Level (ALL-in-One), KC Level (One-by-One). When running the results, there are also various outputs that are generated, may I check if the Questions Level (All-in-One) scenario results is matched to the oriaucxxx results? (xxx is the fusion mechanism)

**Comment by sonyawong** [comment]
> ![Screenshot from 2023-10-04 10-59-38](https://user-images.githubusercontent.com/146905955/272449609-d31649af-3bb2-4274-ac6c-259cde6ebc65.png) I understand that there are various scenarios e.g. Questions Level (All-in-One), KC Level (ALL-in-One), KC Level (One-by-One). When running the results, there are also various outputs that are generated, may I check if the Questions Level (All-in-One) scenario results is matched to the oriaucxxx results? (xxx is the fusion mechanism)

hi, thank you for ur interest in our work. The results ori+auc/acc+xxx(fusion mechanism) are the evaluation results in question-level without slide window setting. For the results of window setting, the results are denotes as window+auc/acc+xxx(fusion mechanism). Moreover, the KC Level test+auc/acc and window_test+auc/acc.

**Reply by cwsy92** [reply]
so, may I know which setting (if any) would be the best to examine in knowledge tracing?

**Comment by Xemorr** [comment]
Hi, what is the reason for breaking after 5000 items in the ednet preprocess?

**Reply by Patrick-bj** [reply]
Because the ednet dataset is too large, due to the training time factor we only intercept part of it, you can control the size of the intercept as needed.

**Comment by Echo628999** [comment]
pykt团队您好，我先后在akt和simplekt上得到了在数据集as2009上auc，发现明显高于论文数据
同时我又在测akt在al2005上的表现，simplekt 和 akt的auc同时高达0.9左右，同样远高于论文数据，
因为数据集都是官方下载，我怀疑是论文里提到的标签泄露问题，请问如何修改，解决这个问题，复现出平台论文里介绍的数据
<img width="1268" alt="1698197207555" src="https://github.com/pykt-team/pykt-toolkit/assets/148762049/03fb6de9-1985-488f-b477-cdd5f779b63d">



<img width="1223" alt="akt al2005" src="https://github.com/pykt-team/pykt-toolkit/assets/148762049/3a78c81c-7942-4caa-94a5-1147f60d0f01">

**Reply by Echo628999** [reply]
我们组想要用pykt完成一些工作，不知道佬能不能留下联系方式邮箱之类的咨询一些问题

**Comment by Patrick-bj** [comment]
> pykt团队您好，我先后在akt和simplekt上得到了在数据集as2009上auc，发现明显高于论文数据 同时我又在测akt在al2005上的表现，simplekt 和 akt的auc同时高达0.9左右，同样远高于论文数据， 因为数据集都是官方下载，我怀疑是论文里提到的标签泄露问题，请问如何修改，解决这个问题，复现出平台论文里介绍的数据 <img alt="1698197207555" width="1268" src="https://user-images.githubusercontent.com/148762049/277846683-03fb6de9-1985-488f-b477-cdd5f779b63d.png">
> 
> <img alt="akt al2005" width="1223" src="https://user-images.githubusercontent.com/148762049/277871194-3a78c81c-7942-4caa-94a5-1147f60d0f01.png">
